pkg/controllers/disruption/types.go

```diff
@@ -90,12 +90,8 @@ func NewCandidate(ctx context.Context, kubeClient client.Client, recorder events
 		recorder.Publish(disruptionevents.Blocked(node.Node, node.NodeClaim, fmt.Sprintf("NodePool %q not found", nodePoolName))...)
 		return nil, fmt.Errorf("nodepool %q can't be resolved for state node", nodePoolName)
 	}
+	// We only care if instanceType in non-empty consolidation to do price-comparison.
 	instanceType := instanceTypeMap[node.Labels()[corev1.LabelInstanceTypeStable]]
```

Comment:
I wonder if we should even populate this as part of building the candidates here or if this should all be encapsulated inside of the consolidation bits because now we have components of our candidate that can be nil (which feels a bit awkward to me) e.g. capacity type, zone, instance type and if you're the caller or we extended this later it's unclear that these are going to be nil or empty when I'm grabbing them

Reply:
You have to pass in a lot of NodePool -> Instance Type -> Offerings map data if you pass it through into the consolidation calls. It's a bit more of an involved change, but I think this generally makes sense to me that you'd  pass in a nil if the instance type doesn't exist. I do agree it expands the possibility that we mess up and get a nil dereference. WDYT

Reply:
I'm fine with a fast-follow -- it definitely doesn't completely encapsulate the behavior right now but I'm also fine not churning on it